enhancements/microshift/update-and-rollback.md

```diff
@@ -0,0 +1,255 @@
+---
+title: update-and-rollback
+authors:
+  - oglok 
+reviewers:
+  - "@copejon"
+  - "@fzdarsky"
+  - "@ggiguash"
+  - "@majopela"
+  - "@dhellmann"
+  - "@zshi-redhat"
+  - "@pacevedom"
+approvers:
+  - "@dhellmann"
+api-approvers:
+  - None
+creation-date: 2023-01-09
+last-updated: 2023-01-09
+tracking-link:
+  - https://issues.redhat.com/browse/USHIFT-518
+---
+
+# MicroShift Upgrades and Rollbacks
+
+## Summary
+
+Allow successful upgrades of MicroShift running on Edge Devices and enable
+the ability of going back to a previous working state. To support this process,
+MicroShift relies on the transactional features of OSTree by atomically
+switching between states.
+
+## Motivation
+
+Managing the life cycle of Edge Devices at scale brings certain challenges
+regarding configuration and maintenance. By leveraging the features of OSTree for
+upgrades and rollbacks, Edge Devices can automatically detect if a new version
+of MicroShift has been provided and upgrade accordingly.
+
+This enhancenment covers the mechanisms to save the state of a running device
+by performing a backup of all required MicroShift resources and restoring it in
+the case of a broken state.
+
+### User Stories
+
+As an application developer, I want to focus on releasing new stable versions of
+my application that work in production.
+
+As an application deployer, I want to focus only on the lifecycle management for
+my application and forget about how MicroShift is upgraded.
+
+As an application deployer, I can provide new versions of my application as part
+of the OSTree, and enforce its upgrade
+
+As a device administrator, I can provide a new version of the Operating System
+OSTree and the edge device will upgrade automatically.
+
+As a device administrator, I need to ensure a MicroShift upgrade works and a
+rollback mechanism is in place in case something goes wrong.
+
+### Goals
+
+The goals of this proposal are:
+- How to implement a successful MicroShift upgrade on systems based on `rpm-ostree`
+- Describe the process of creating a MicroShift's data backup
+- Identify what files need to be saved as part of a MicroShift's data backup
+- Describe the process of rolling back to a previous working state
+
+### Non-Goals
+
+The non-goals of this proposal are:
+- How to implement MicroShift healthchecks using `greenboot`. This is covered by
+[this](https://github.com/openshift/enhancements/pull/1306) other enhancement proposal
+- How to implement upgrades of 3rd party applications running on MicroShift
+- How to implement rollbacks of 3rd party applications running on MicroShift
+- How to implement rollbacks without rpm-ostree
+
+## Proposal
+
+### Workflow Description
```

Comment:
We have a few different threads talking about what triggers updates and how much MicroShift is involved in that process. One way to clear that up would be to describe the workflow in this section in active voice using explicit actors at each step. (Instead of "a new image shall be created" say "the administrator will build a new image", for example.) We could keep it simple by assuming a user is doing some steps that might be automated when running at scale, and then use a variation to describe which steps could be automated.